src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -386,7 +388,16 @@ class Crystal::Doc::Generator
       filename = filename[1..-1] if filename.starts_with? File::SEPARATOR
       filename = filename[4..-1] if filename.starts_with? SRC_SEP
 
-      locations << RelativeLocation.new(filename, location.line_number, url)
+      show_line_number = locations.any? do |location|
+        if location.filename == filename
+          location.show_line_number = true
```

Comment:
How does this even work? `location` is a struct.

Reply:
Thats because of the `setter` I added for `show_line_number` in line 364.

Reply:
But it's a **struct**. When you do `locations.any? do |location|`, that `location` is a copy of the struct saved in the array, so it shouldn't be modified. Are you sure this is working? Did you compile the compiler and ran the docs?

Reply:
(we need specs for the docs generator)

Reply:
I'm not sure if this is a good idea. Making it a class adds lots of heap allocations. Why not just change the code to work with a struct? `locations` is an array and instead of using `any?` you can just loop through it with `each_with_index` and store the updated entry back into the array.